### DISCUSSION TITLE
How to observe key strokes?

### DISCUSSION THREAD

**Discussion by ashebanow** [Aylur/ags#531] [Q&A] [answered]
I'd like to make a panel that monitors the keystrokes and displays them graphically. Something like [screenkey](https://gitlab.com/screenkey/screenkey), but hyprland aware so it displays the command's description as well.

I know I can create a `Window` with `keymode: exclusive` and that will steal all keyboard input, and that I can bind to the Hyprland service to get notified when the keyboard submap changes.

But I don't really want to steal input, i just want to observe it. Is there a way to do that in ags?

**Comment by Aylur** [comment, marked as answer]
not directly no, you would have to use some other utility like libinput and run that as a subprocess
